bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -349,17 +339,43 @@ private static HSBType rgbwToHsb(PercentType[] rgbw) {
      * @return the corresponding {@link HSBType}.
      * @throws IllegalArgumentException when input array has wrong size or exceeds allowed value range.
      */
-    public static HSBType rgbToHsb(PercentType[] rgb) throws IllegalArgumentException {
-        if (rgb.length == 4) {
-            return rgbwToHsb(rgb);
-        }
-        if (rgb.length != 3) {
-            throw new IllegalArgumentException("RGB array needs exactly three values!");
+    public static HSBType rgbToHsb(PercentType[] rgbw) throws IllegalArgumentException {
+        if (rgbw.length < 3 || rgbw.length > 4) {
+            throw new IllegalArgumentException("rgbToHsb() requires 3 or 4 arguments");
         }
 
-        BigDecimal r = rgb[0].toBigDecimal();
-        BigDecimal g = rgb[1].toBigDecimal();
-        BigDecimal b = rgb[2].toBigDecimal();
+        BigDecimal r;
+        BigDecimal g;
+        BigDecimal b;
+
+        if (rgbw.length < 4) {
```

Comment:
If you compare `== 3` it's a bit more clear when this branch is selected (length == 0, 1, 2 are not possible due to the check before)

Reply:
Ok. Done.